src/main/java/core/basesyntax/handler/OperationStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.handler;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return operationHandlerMap.get(operation);
```

Comment:
Consider adding a check to handle cases where the `operationHandlerMap` does not contain a handler for the given operation. This could prevent potential `NullPointerException` when the handler is used.